include/rcpputils/asserts.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__ASSERTS_HPP_
+#define RCPPUTILS__ASSERTS_HPP_
+
+#include <exception>
+#include <stdexcept>
+#include <string>
+
+#include "rcpputils/visibility_control.hpp"
+
+// Needed to disable compiler warning for exporting a class that extends a
+// non-DLL-interface class.
+// This should be fine since its extending an STL class.
+#ifdef _WIN32
+# pragma warning(push)
```

Comment:
Don't you need to pop this at some point?

Reply:
yes.